explainer.md

```diff
@@ -89,21 +89,25 @@ The second mode where styles are copied to a container element won't be part of
 
 ### Nested transition elements
 
-In the example transition, the content of the header cross-fades from Page-A to Page-B. An even smoother transition could be achieved by also animating the site title and avatar 'chip' independently. However, 'v1' of this proposal disallows a shared element to be nested inside another shared element.
+In the example transition, the content of the header cross-fades from Page-A to Page-B. An even smoother transition could be achieved by also animating the site title and avatar 'chip' independently. When a developer offers nested shared elements, the state can be captured in the following two modes :
 
-The restriction avoids the need to preserve the hierarchy of shared elements and associated properties (transform, clip, effects inherited by descendants) in the DOM representation created to render the images referenced above. This helps in minimizing the scope for 'v1' of the feature.
+### Flat List Mode
+The content of all nested shared elements is removed when generating the image for a shared element so they can be animated as independent images. This is similar to removing the content of shared elements from the root image.
 
-- [Open question](https://github.com/WICG/shared-element-transitions/issues/74): If the developer tries to do this, what should happen, and when is that verified? Is the nested offering ignored? Is the whole transition abandoned?
+In addition to the element's image, the browser also caches the element's border-box bounds and a transform positioning the element relative to the viewport so that it overlaps exactly with its quad on the old page.
+
+## Hierarchy Mode
+In this mode, the hierarchy of shared elements is preserved along with a transform which is relative to its nearest ancestor shared element instead of the viewport. When used with the "computed style + content image" mode, this also preserves other hierarchical properties (clip, effects inherited by descendants) that a shared element inherits from its ancestor. 
+
+This allows nested shared elements to continue to be positioned and clipped relative to their ancestor shared elements during a transition. This is proposed as a separate mode to minimize the scope for 'v1' of the feature. See [issue](https://github.com/WICG/shared-element-transitions/issues/74) for detailed discussion.
```

Comment:
Is this a separate mode solely for minimizing the scope of the v1?

Reply:
That's the main reason, yes. Other reason is that we can choose to not support it for the out of process cross-origin implementation. But the primary motivation is to scope it out of v1.

Reply:
Would it be possible to support this cross-origin if hierarchical info were sent to wherever it is that interpolation happens?

Reply:
Yeah, it's definitely possible. We'll need to build a data structure that preserves the transform, clip and stacking context hierarchy of these elements. All of those should be the same since with paint containment each shared element creates a stacking context anyway.

One aspect worth pointing out is that this hierarchy mode will only retain positioning and clipping from an ancestor. If the ancestor has effects like opacity (which should continue to be inherited by the descendants) then we have to ensure that the opacity is "live" and not baked into the snapshot. So it applies to the shared element's image and also to the images of its descendants.

@chrishtr has also mentioned not painting visual effects into the snapshot. This is a divergence from the element() function behaviour but maybe it makes sense to have the "As a single image" mode be "As style for visual effects + single image" mode instead. The styles for visual effects which apply to descendants are copied to the container.

We can support "As style for visual effects + single image" + hierarchy for cross-origin but supporting "As the element's computed style + content image" for cross-origin will be too difficult.

Reply:
I've filed a WICG issue for the question about visual effects above.

Reply:
Thanks for the extra info. I was just hoping to clarify the point raised earlier and didn't mean to hold up the review. I will stamp and defer to Vlad's review.

Reply:
"This is proposed as a separate mode to minimize the scope for 'v1' of the feature." This makes it sound like separating into two modes minimizes the scope. Can we clarify that only Flat List mode will be targeted for v1 to minimize scope.

Reply:
@jakearchibald mentioned that he'll reword this section to make the details for the 2 modes clearer. Will make sure that flat list being the only mode for v1 is obvious after his edits.